clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/AggKafkaConsumerServiceStats.java

```diff
@@ -0,0 +1,123 @@
+package com.linkedin.davinci.stats;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.meta.ReadOnlyStoreRepository;
+import com.linkedin.venice.stats.AbstractVeniceAggStoreStats;
+import com.linkedin.venice.stats.StatsSupplier;
+import com.linkedin.venice.utils.SystemTime;
+import io.tehuti.metrics.MetricsRepository;
+import java.util.function.LongSupplier;
+
+
+/**
+ * This class is an aggregate place that keeps stats objects for multiple stores and total stats for each region for
+ * AggKafkaConsumerService.
+ *
+ * For total stats for a given region, use this class to record stats. For store-level stats, delegate them to
+ * {@link KafkaConsumerServiceStats}.
+ */
+public class AggKafkaConsumerServiceStats extends AbstractVeniceAggStoreStats<KafkaConsumerServiceStats> {
+  public AggKafkaConsumerServiceStats(
+      String regionName,
+      MetricsRepository metricsRepository,
+      ReadOnlyStoreRepository metadataRepository,
+      LongSupplier getMaxElapsedTimeSinceLastPollInConsumerPool,
+      boolean isUnregisterMetricForDeletedStoreEnabled) {
```

Comment:
May I ask why we are passing this boolean everywhere? Could you point out where it is used?

Reply:
This boolean is passed all the way to `AbstractVeniceAggStoreStats` such that when a store is deleted, its sensors would be unregistered.
https://github.com/linkedin/venice/blob/54a68c15756c6b0347be28163bf2c1bc9155077f/internal/venice-common/src/main/java/com/linkedin/venice/stats/AbstractVeniceAggStoreStats.java#L57

Reply:
Oh I think I get it... you move this from AbstractVeniceAggStoreStats right? I see..